internal/services/network/public_ip_resource.go

```diff
@@ -54,24 +53,9 @@ func publicIp() *pluginsdk.Resource {
 
 			"resource_group_name": commonschema.ResourceGroupName(),
 
-			"public_ip_address_allocation": {
-				Type:             schema.TypeString,
-				Optional:         true,
-				Computed:         true,
-				DiffSuppressFunc: suppress.CaseDifference,
-				ExactlyOneOf:     []string{"public_ip_address_allocation", "allocation_method"},
-				Deprecated:       "the `public_ip_address_allocation` property has been renamed to `allocation_method`",
-				ValidateFunc: validation.StringInSlice([]string{
-					string(network.Dynamic),
-					string(network.Static),
-				}, true),
-			},
```

Comment:
was this not supported int 0.10 of the provider? we can't just remove a property like that without first deprecating it

Reply:
if we are going to make hard breaking changes like this in 1.0 of the provider we need to give it some care and make sure existing users are prepared and ok with it

Reply:
Done, thanks for the comment

Reply:
we will have to deprecate this in a 0.11 release, but that can be handled prior to relerase 1.0